src/component/Logic/Logic.js

```diff
@@ -0,0 +1,94 @@
+export default class Logic {
+    constructor(generation = 0, CellsOn = new Map()) {
+      this.generation = generation;
+      this.CellsOn = CellsOn;
+      this.nextGeneration = new Map();
+      this.CellsOff = new Map();
+    }
+  
+    getGeneration() {
+      return this.generation;
+    }
+  
+    getCellsOn() {
+      return this.CellsOn;
+    }
+  
+    addCell(position) {
+      this.CellsOn.set(position.x + " , " + position.y, {x: position.x, y: position.y});
+    }
+  
+    removeCell(position) {
+      this.CellsOn.delete(position); //-> undefined 
+    }
+  
+    isCellOn(position) {
+      return this.CellsOn.has(position); //return boolean
+    }
+  
+    storeCell(position) {
+      if(this.isCellOn(position.x + " , " + position.y)) {
+        this.removeCell(position.x + " , " + position.y);
+      } else {
+        this.addCell(position);
+      }
+  
+      return new Logic(this.generation, this.CellsOn);
+    }
+  
+    addGeneration(){
```

Comment:
What's the need for `generation` and difference between a `generation` and `CellsOn`? This class is tricky to grasp by reading the code.